In [1]:
# Initial Settings.

import os, inspect
import sqlite3
import pandas as pd

path = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
os.chdir(path)

In [2]:
# Connect DB

conn = sqlite3.connect('DB_ver_1.1.db')  # Database 연결 (없는 경우 생성까지)
c = conn.cursor()  # you can create a Cursor object and call its (데이터 추출 뿐 아니라 입력, 수정 등에 사용하기 위한 메소드)

In [3]:
df = pd.read_sql_query("select * from econ;", conn)  # SQL 쿼리 실행하는 경우

In [4]:
df.head()

index      date   ticker        value
0      0  20190411  EIADIST  128053.0000
1      1  20190411  TKBLRON      25.5000
2      2  20190411  CNCOMMF       1.7477
3      3  20190411  LCBBASE       0.7500
4      4  20190411  JPCALLT      -0.1000

< Sample Queries >

SELECT * FROM econ;

SELECT date FROM econ;

SELECT date dt FROM econ;

SELECT * FROM econ WHERE date = 20190411;

SELECT * FROM econ
WHERE date > 20190801 AND ticker = 'USFEFRH';

SELECT * FROM econ 
WHERE ticker = 'USFEFRH' ORDER BY date DESC;

SELECT * FROM econ
WHERE date > 20190801 AND ticker = 'USFEFRH'
ORDER BY date DESC;

SELECT date, ticker, value FROM econ
WHERE date > 20190801 AND ticker = 'USFEFRH'
ORDER BY date DESC;

In [5]:
qry = "SELECT date, ticker, value FROM econ WHERE date > 20190801 AND ticker = 'USFEFRH' ORDER BY date DESC;"

df = pd.read_sql_query(qry, conn)
df.head()

date   ticker  value
0  20190829  USFEFRH   2.25
1  20190828  USFEFRH   2.25
2  20190827  USFEFRH   2.25
3  20190826  USFEFRH   2.25
4  20190823  USFEFRH   2.25

In [6]:
qry = "SELECT date, ticker, value FROM econ WHERE date > 20190801 AND ticker IN ('USFEFRH', 'USFEFRL') ORDER BY date DESC;"

df = pd.read_sql_query(qry, conn)
df.head()

date   ticker  value
0  20190829  USFEFRH   2.25
1  20190829  USFEFRL   2.00
2  20190828  USFEFRL   2.00
3  20190828  USFEFRH   2.25
4  20190827  USFEFRH   2.25

In [7]:
df = df.pivot(index = 'date', columns = 'ticker', values = 'value')
print(df)

ticker    USFEFRH  USFEFRL
date                      
20190802     2.25      2.0
20190805     2.25      2.0
20190806     2.25      2.0
20190807     2.25      2.0
20190808     2.25      2.0
20190809     2.25      2.0
20190812     2.25      2.0
20190813     2.25      2.0
20190814     2.25      2.0
20190815     2.25      2.0
20190816     2.25      2.0
20190819     2.25      2.0
20190820     2.25      2.0
20190821     2.25      2.0
20190822     2.25      2.0
20190823     2.25      2.0
20190826     2.25      2.0
20190827     2.25      2.0
20190828     2.25      2.0
20190829     2.25      2.0


In [ ]:
df =        # "BETWEEN 시작지점 AND 종료지점" 과 같은 형태의 쿼리 이용
              # 2019년 6~8월의 미국 기준금리 상단 및 하단을 최근 데이터가 위로 오도록 추출 (ticker = "USFEFRH", "USFEFRL")

print(df)

더 다양한 쿼리들은 아래 Tutorial 에서 직접 연습 가능

https://www.sqlitetutorial.net/